main.cpp

```diff
@@ -121,369 +155,446 @@ struct CarWash
-5 properties:
-    1) number of octaves 
-    2) number of MIDI channel
-    3) keyboard size in pixel 
-    4) key colours
-    5) velocity range 
-3 things it can do:
-    1) send midi notes 
-    2) display MIDI numbers (note to MIDI)
-    3) recieve input from computer keyboard
- */
+void DawMixer::recordReady(){}
+
+void DawMixer::panSnd(float pan)
+{
+    panPos += pan;
+}
+
+void DawMixer::audioVolume (float laudioLvl, float normalize)
+{
+    lvl = laudioLvl * normalize;
+}
 
 struct VirtualMidiKey
 {
-    //number of octaves
     int numOctaves = 6; 
-    //number of MIDI channel
     int numMidiChanels = 32;
-    //keyboard size in pixel
     int keyboardSize = 500; 
-    //key colours
     bool invertKeyColour = false;
-    //velocity range
     int velocityRange = 127;
 
-    //send midi notes 
     void sendMidi (int keyIndex, int velocity);
-    //display MIDI numbers (note to MIDI)
     int displayMidi (char noteName);
-    //recieve input from computer keyboard
     void receiveInputKey (int inputNum, int midiChanel);
 };
 
-/*
-Thing 7) Toolbar Docker
-5 properties:
-    1) number of tools in docker
-    2) tool selector size  
-    3) docker position 
-    4) docker size
-    5) number of row
-3 things it can do:
-    1) change tool for editing
-    2) customize displayed tools
-    3) activate metronom
- */
+void VirtualMidiKey::sendMidi(int keyIndex, int velocity)
+{
+    keyboardSize = keyIndex * numOctaves;
+    velocityRange += velocity;
+}
+
+int VirtualMidiKey::displayMidi(char noteName)
+{
+    return noteName;
+}
+
+void VirtualMidiKey::receiveInputKey(int inputNum, int midiChanel)
+{
+    numMidiChanels -= (inputNum + midiChanel);
+}
 
 struct ToolbarDocker
 {
-    //number of tools in docker
     int numDockerTools = 12;
-    //tool selector size in pixels 
     float toolSelectorSize = 25.5f;
-    //docker position 
     float dockerPosX = 255.0f;
-    //docker size
     float dockerWidth = 500.0f;
-    //number of row
     int numDockerRow = 2;
 
-    //change tool for editing
     void changeTool (int toolListIndex, int replaceIndex);
-    //customize displayed tools
-    void customizeTool (int indexSwitch, int newToolIndex);
-    //activate metronom
-    void activateMetronom (bool mentroPlay);
-
+    int customizeTool (int indexSwitch, int newToolIndex);
+    bool activateMetronom (bool metroPlay);
 };
 
-/*
-Thing 8) Performance Meter
-5 properties:
-    1) text size 
-    2) window size
-    3) font style
-    4) background colour
-    5) CPU graph clock speed 
-3 things it can do:
-    1) display CPU
-    2) display RAM use
-    3) display Disk read in Bytes/s
- */
+void ToolbarDocker::changeTool(int toolListIndex, int replaceIndex)
+{
+    numDockerTools = toolListIndex + replaceIndex;   
+}
+
+int ToolbarDocker::customizeTool(int indexSwitch, int newToolIndex)
+{
+    return indexSwitch - newToolIndex;
+}
+
+bool ToolbarDocker::activateMetronom(bool metroPlay)
+{
+   bool state = 0;
```

Comment:
Never initialize booleans with integers.  always initialize them with `true` or `false`.
The same rule applies when you're setting their value.  Fix this, and the lines below where you assign to `state`
- [ ] revise